### DISCUSSION TITLE
Services going down every 5 mins or so

### DISCUSSION THREAD

**Discussion by rxerium** [telekom-security/tpotce#1445] [Q&A]
Hi there,

Some services keep restarting / going down intermittently on my cloud VPS running 16GB RAM / 6vCPU cores / 400GB SSD. See the below information which might be helpful. 

<a name="info"></a>
## ⚠️ Basic support information (commands are expected to run as `root`)

- What version of the OS are you currently using `lsb_release -a` and `uname -a`?
Ubuntu 22.04.3 LTS

- What T-Pot version are you currently using?
I just installed it so the latest version. 

- What edition (Standard, Nextgen, etc.) of T-Pot are you running?
Standard

- What architecture are you running on (i.e. hardware, cloud, VM, etc.)?
Cloud VPS

- Did you have any problems during the install? If yes, please attach `/install.log` `/install.err`.
Yes when running `./install.sh` I get `Aborting. Debian jammy is not supported` so I modified the installer. See below.

- How long has your installation been running?
1 hour 

- Did you install upgrades, packages or use the update script?
Prior to running the script yes

- Did you modify any scripts or configs? If yes, please attach the changes.
Yes, changed `mySUPPORT="FALSE" ` to `mySUPPORT="TRUE"` in file ./install.sh line 364

- Please provide a screenshot of `glances` and `htop`.
![image](https://github.com/telekom-security/tpotce/assets/59293085/667f6bef-6cc3-42c2-8b4f-28eead926f15)
![image](https://github.com/telekom-security/tpotce/assets/59293085/bff8f680-22b9-4c00-a641-a9d262009ac5)

- How much free disk space is available (`df -h`)?
360GB

- What is the current container status (`dps.sh`)?
![image](https://github.com/telekom-security/tpotce/assets/59293085/40828db2-cfa0-40b2-a2ac-9f9b5a8f54e8)

- What is the status of the T-Pot service (`systemctl status tpot`)?
![image](https://github.com/telekom-security/tpotce/assets/59293085/dcb33c46-8356-4035-a7cd-ec132900ef0b)

- What ports are being occupied? Stop T-Pot `systemctl stop tpot` and run `netstat -tulpen`
![image](https://github.com/telekom-security/tpotce/assets/59293085/909eebea-c5bd-4b15-a4f9-27962a570c1d)

- If a single container shows as `DOWN` you can run `docker logs <container-name>` for the latest log entries
![image](https://github.com/telekom-security/tpotce/assets/59293085/47612dee-2516-441d-bd98-d5a2a33637b9)

**Comment by rxerium** [comment]
~~Okay I think its working~~, but should I have any more services available to me or is this everything? Looks like im missing Kibana
![image](https://github.com/telekom-security/tpotce/assets/59293085/cbc65a09-f6da-4017-8f71-e1fb6d849848)

Edit: It keeps going down again. I've ran some intrusive scans against the honeypot and the map does not pick it up.

**Comment by rxerium** [comment]
Hey all, any updates here? Happy to provide more details where necessary

**Reply by cah117** [reply]
This happened to me just today after editing the Cowrie docker-compose file and running `docker-compose up -d --no-deps --force-recreate` within the `/opt/tpot/docker/cowrie/ `directory. This broke everything BUT Cowrie it seems, because it is now the ONLY container that doesn't restart every 2 minutes.

The solution I found: Run `update.sh -y` from `/opt/tpot/` which resolved the problems by reverting all of the scripts and docker-compose files back to their defaults. All old logs were still there from before the services starting crashing.

Edit - Disregard this:
_The Cowrie container is read-only by default in the docker-compose file. But if you make it writable to add your own files to fs.pickle and rerun docker-compose, that breaks everything else it seems because that was literally the only change I made on a second test of this. So, yeah, I'm not sure how you're supposed to customize the Cowrie honeypot if you can't write to it, either within the container directly or through `docker cp`._

I see now, I guess I should have been updating the docker compose section for Cowrie at `/opt/tpot/etc/tpot.yml` instead.
